### ISSUE TITLE
deprecationwarning: isDefaultPage

### ISSUE THREAD

**Issue by spereverde** [collective/collective.searchandreplace#57]
when running a Plone 5.2 (python 3.6) setup we get a deprecation warning:
````
/buildout/eggs/collective.searchandreplace-8.2.1-py3.6-linux-x86_64.egg/collective/searchandreplace/browser/searchreplaceform.py:9: DeprecationWarning: isDefaultPage is deprecated. Import from Products.CMFPlone instead
  from plone.app.layout.navigation.defaultpage import isDefaultPage
````
todo: adjust import for different versions of plone

**Comment by spereverde** [comment]
I also found this:
https://github.com/plone/plone.portlet.collection/commit/937d93705f442e364a8f1512d53300fc41486e9f
I can make a PR but would like to know if we want to import just:
`from Products.CMFPlone.defaultpage import is_default_page`
or keep the backward compatibility with Plone 4 with a try except
any recommendations?

**Comment by mauritsvanrees** [comment]
In Plone 6 it should be:

    from plone.base.defaultpage import is_default_page

Perhaps a try/except that works in Plone 4/5/6?

But for most add-ons it makes sense to only support Plone 5.2+, and maybe only Python 3. Would be fine for me.

**Comment by Rudd-O** [comment]
https://github.com/Rudd-O/collective.searchandreplace/commit/f2d6a190269041d3d80148b07440438a4a2505bf can be cherry-picked for the fix.

**Comment by spereverde** [comment]
cherry-picked in https://github.com/collective/collective.searchandreplace/commit/cf438c22707df1d6cd7266c273d454378432d900

**Comment by Rudd-O** [comment]
Hopefully there will be a release before Plone 6 GA.

**Comment by mauritsvanrees** [comment]
This was was finally released on March 10 2023, in version 8.2.2.
